sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -384,17 +385,28 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
     val bis = new ByteArrayInputStream(bytes)
     val ins = new DataInputStream(codec.compressedInputStream(bis))
 
-    new Iterator[InternalRow] {
+    new NextIterator[InternalRow] {
       private var sizeOfNextRow = ins.readInt()
-      override def hasNext: Boolean = sizeOfNextRow >= 0
-      override def next(): InternalRow = {
-        val bs = new Array[Byte](sizeOfNextRow)
-        ins.readFully(bs)
-        val row = new UnsafeRow(nFields)
-        row.pointTo(bs, sizeOfNextRow)
-        sizeOfNextRow = ins.readInt()
-        row
+      override def getNext(): InternalRow = {
```

Comment:
Sorry last comment. Can we define this func as:

```scala
override def getNext(): InternalRow = if (sizeOfNextRow >= 0) {
  ...
}
```

? I think that will reduce the diff.

Reply:
Or turn the existing func `override def next(): InternalRow = {` to a private function, and call it in `getNext`.

Reply:
It's minor but I am suggesting to make the backport easier to reduce conflicts + make it easier to review.